packages/shared/src/hooks/useFeed.ts

```diff
@@ -112,21 +113,33 @@ export default function useFeed<T>(
     settings,
     shouldShowSurvey,
     showPublicSquadsEligibility,
+    onEmptyFeed,
   } = params;
   const { user, tokenRefreshed } = useContext(AuthContext);
   const queryClient = useQueryClient();
   const isFeedPreview = feedQueryKey?.[0] === RequestKey.FeedPreview;
 
   const feedQuery = useInfiniteQuery<FeedData>(
     feedQueryKey,
-    ({ pageParam }) =>
-      request(graphqlUrl, query, {
+    async ({ pageParam }) => {
+      const res = await request(graphqlUrl, query, {
         ...variables,
         ...(getShouldRefreshFeed() && { refresh: true }),
         first: pageSize,
         after: pageParam,
         loggedIn: !!user,
-      }),
+      });
+
+      if (
+        !feedQuery?.data?.pages[0]?.page.edges.length &&
+        !res?.page?.edges?.length &&
+        onEmptyFeed
+      ) {
+        onEmptyFeed();
+      }
```

Comment:
Avoids usage of `useEffect` and makes test easier to manage.

Reply:
Oh I see now :) 
Was this one of the few places?

Reply:
Yes, should ensure executes only once and avoids the use effect 🥳